src/main.js

```diff
@@ -76,9 +76,15 @@ export function generateCompress(isEncode) {
  * @param {Object} options
  * @param {string} [options.body]
  */
-export const fetchToCurl = (url, options) => {
+const fetchToCurl = (requestInfo, requestInit) => {
+  const url = typeof requestInfo === 'string' ? requestInfo : requestInfo.url;
+  const options = typeof requestInfo === 'string' ? requestInit : requestInfo;
+  const fetchHeaders = options.headers ?
+    (typeof options.headers.forEach === 'function' ? options.headers : new Headers(options.headers)) :
```

Comment:
I think we should move this into the `fetchHeaders` function and handle different inputs there.  The handling for `null` and `undefined` is already in there. Also it makes testing easier.

Maybe it would be best to just always call `new Headers(inHeaders)` in `generateHeader`. (The constructor of Headers can deal with Objects and Headers)